### ISSUE TITLE
Tensor sizes error when testing a modified YOLOv5 nano model (.yaml)

### ISSUE THREAD

**Issue by lessleeb** [ultralytics/yolov5#6203]
### Search before asking

- [X] I have searched the YOLOv5 [issues](https://github.com/ultralytics/yolov5/issues) and [discussions](https://github.com/ultralytics/yolov5/discussions) and found no similar questions.

### Question

Hi, I am trying to modify  YOLOv5 nano model. In this case I eliminated the head P5, as shown below in the .yaml file. (it's actually commented).
Additionally, since I am using only 2 anchors now, I commented the corresponding one (P5), and modified the detection part as well. Although, in the training part I do not have any problem, the detection part shows an error regarding the tensor sizes. I have been trying to find what could be the problem, but I've not been able to find it.
This only happens when I remove a layer from the model. If I replace a layer (ex: a Conv for a Focus layer) there is not problem in the detection part.
Could you tell me what could be the problem? Do I need to modify anything else in the code?
Here is the modified yaml file.

YOLOv5n.YAML

# Parameters
nc: 80  # number of classes
depth_multiple: 0.33  # model depth multiple
width_multiple: 0.25  # layer channel multiple

anchors:
  - [10,13, 16,30, 33,23]  # P3/8
  - [30,61, 62,45, 59,119]  # P4/16
  #- [116,90, 156,198, 373,326]  # P5/32

# YOLOv5 v6.0 backbone
backbone:

  [[-1, 1, Conv, [64, 6, 2, 2]],  # 0-P1/2
   [-1, 1, Conv, [128, 3, 2]],  # 1-P2/4
   [-1, 3, C3, [128]],
   [-1, 1, Conv, [256, 3, 2]],  # 3-P3/8
   [-1, 6, C3, [256]],
   [-1, 1, Conv, [512, 3, 2]],  # 5-P4/16
   [-1, 9, C3, [512]],
   [-1, 1, Conv, [1024, 3, 2]],  # 7-P5/32
   [-1, 3, C3, [1024]],
   [-1, 1, SPPF, [1024, 5]],  # 9
  ]

# YOLOv5 v6.0 head
head:
  [[-1, 1, Conv, [512, 1, 1]],
   [-1, 1, nn.Upsample, [None, 2, 'nearest']],
   [[-1, 6], 1, Concat, [1]],  # cat backbone P4
   [-1, 3, C3, [512, False]],  # 13

   [-1, 1, Conv, [256, 1, 1]],
   [-1, 1, nn.Upsample, [None, 2, 'nearest']],
   [[-1, 4], 1, Concat, [1]],  # cat backbone P3
   [-1, 3, C3, [256, False]],  # 17 (P3/8-small)

   [-1, 1, Conv, [256, 3, 2]],
   [[-1, 14], 1, Concat, [1]],  # cat head P4
   [-1, 3, C3, [512, False]],  # 20 (P4/16-medium)

   [[17, 20], 1, Detect, [nc, anchors]],  # Detect(P3, P4, P5)
  ]

And this is the error:
**Exception has occurred: RuntimeError**
torch.cat(): Sizes of tensors must match except in dimension 1. Got 26 and 25 in dimension 2 (The offending index is 1)
  File "C:\Users\Desktop\yolov5-20211118\models\common.py", line 274, in forward
    return torch.cat(x, self.d)
  File "C:\Users\Desktop\yolov5-20211118\models\yolo.py", line 150, in _forward_once
    x = m(x)  # run
  File "C:\Users\Desktop\yolov5-20211118\models\yolo.py", line 127, in forward
    return self._forward_once(x, profile, visualize)  # single-scale inference, train
  File "C:\Users\Desktop\yolov5-20211118\models\common.py", line 356, in forward
    y = self.model(im) if self.jit else self.model(im, augment=augment, visualize=visualize)
  File "C:\Users\Desktop\yolov5-20211118\detect.py", line 115, in run
    pred = model(im, augment=augment, visualize=visualize)
  File "C:\Users\Desktop\yolov5-20211118\detect.py", line 240, in main
    run(**vars(opt))
  File "C:\Users\Desktop\yolov5-20211118\detect.py", line 245, in <module>
    main(opt)

**Comment by glenn-jocher** [comment]
@lessleeb this allows you to create a P3, P4 output YOLOv5s model. Large object performance will be reduced however.

```yaml
# YOLOv5 🚀 by Ultralytics, GPL-3.0 license

# Parameters
nc: 80  # number of classes
depth_multiple: 0.33  # model depth multiple
width_multiple: 0.50  # layer channel multiple
anchors: 3

# YOLOv5 v6.0 backbone
backbone:
  # [from, number, module, args]
  [[-1, 1, Conv, [64, 6, 2, 2]],  # 0-P1/2
   [-1, 1, Conv, [128, 3, 2]],  # 1-P2/4
   [-1, 3, C3, [128]],
   [-1, 1, Conv, [256, 3, 2]],  # 3-P3/8
   [-1, 6, C3, [256]],
   [-1, 1, Conv, [512, 3, 2]],  # 5-P4/16
   [-1, 9, C3, [512]],
   [-1, 1, Conv, [1024, 3, 2]],  # 7-P5/32
   [-1, 3, C3, [1024]],
   [-1, 1, SPPF, [1024, 5]],  # 9
  ]

# YOLOv5 v6.0 head
head:
  [[-1, 1, Conv, [512, 1, 1]],
   [-1, 1, nn.Upsample, [None, 2, 'nearest']],
   [[-1, 6], 1, Concat, [1]],  # cat backbone P4
   [-1, 3, C3, [512, False]],  # 13

   [-1, 1, Conv, [256, 1, 1]],
   [-1, 1, nn.Upsample, [None, 2, 'nearest']],
   [[-1, 4], 1, Concat, [1]],  # cat backbone P3
   [-1, 3, C3, [256, False]],  # 17 (P3/8-small)

   [-1, 1, Conv, [256, 3, 2]],
   [[-1, 14], 1, Concat, [1]],  # cat head P4
   [-1, 3, C3, [512, False]],  # 20 (P4/16-medium)

   [[17, 20], 1, Detect, [nc, anchors]],  # Detect(P3, P4)
  ]
```

**Comment by lessleeb** [comment]
Hi @glenn-jocher . Thank you for your response.
I think you meant _**anchors : 2**_, right? Since I am using only P3, P4 outputs. 
I have modified the .yaml file as you indicated here, and I am still having the same error. Let me explain it.
Train:
Since I am using a customized dataset.
I trained using the pretrained weights  **weights, default=ROOT / 'yolov5n.pt'**
and, **cfg, default='models/yolov5n.yaml**
Detect:
I used the weights obtained after training.
**weights, default=ROOT / 'runs/train/exp/weights/last.pt'**

I think when training the model is saved along with the weights, so I don't understand what is the problem when running the detection. 
![image](https://user-images.githubusercontent.com/97152706/148356408-eac5df2e-3b93-4fcd-b681-a3c09fea88b6.png)

**Comment by glenn-jocher** [comment]
@lessleeb I created a PR with a -34 model and am able to reproduce your error above now. Seems related to minimum stride constraints dropping from 32 to 16 on this custom model. Might need to enforce a floor of 32, I'll take a look.

<img width="1656" alt="Screen Shot 2022-01-06 at 10 07 52 AM" src="https://user-images.githubusercontent.com/26833433/148429874-5297b100-04e5-484c-9870-5cda9a9779c4.png">

**Comment by glenn-jocher** [comment]
@lessleeb good news 😃! Your original issue may now be fixed ✅ in PR #6230. This PR fixes your original issue and also creates an official P3, P4 only output config `yolov5-p34.yaml` in models/hub. A quick training and inference verification passes without error:
<img width="1412" alt="Screen Shot 2022-01-06 at 11 04 45 AM" src="https://user-images.githubusercontent.com/26833433/148437724-9fc94d9d-a410-4691-b6cc-c2fe9011a1ad.png">

To receive this update:
- **[Git](https://github.com/ultralytics/yolov5)** – `git pull` from within your `yolov5/` directory or `git clone https://github.com/ultralytics/yolov5` again
- **[PyTorch Hub](https://pytorch.org/hub/ultralytics_yolov5/)** – Force-reload `model = torch.hub.load('ultralytics/yolov5', 'yolov5s', force_reload=True)`
- **[Notebooks](https://github.com/ultralytics/yolov5/blob/master/tutorial.ipynb)** – View updated notebooks  <a href="https://colab.research.google.com/github/ultralytics/yolov5/blob/master/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolov5"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **[Docker](https://hub.docker.com/r/ultralytics/yolov5)** – `sudo docker pull ultralytics/yolov5:latest` to update your image <a href="https://hub.docker.com/r/ultralytics/yolov5"><img src="https://img.shields.io/docker/pulls/ultralytics/yolov5?logo=docker" alt="Docker Pulls"></a>

Thank you for spotting this issue and informing us of the problem. Please let us know if this update resolves the issue for you, and feel free to inform us of any other **issues** you discover or **feature requests** that come to mind. Happy trainings with YOLOv5 🚀!

**Comment by lessleeb** [comment]
Thank you! I have updated and tested YOLO in my environment. 
There is not error anymore.
